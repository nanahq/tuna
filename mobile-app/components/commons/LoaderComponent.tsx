import {View, ActivityIndicator, ActivityIndicatorProps, ViewStyle, StyleProp} from 'react-native'
import {tailwind} from "@tailwind";

interface LoaderComponentProps extends ActivityIndicatorProps {
 containerStyle?: StyleProp<ViewStyle>
}
export function LoaderComponent (props: LoaderComponentProps): JSX.Element {
    const {containerStyle, ...otherProps} = props
    return (
        <View style={[tailwind(''), containerStyle]}>
           <ActivityIndicator {...otherProps} />
        </View>
    )
}


export function LoaderComponentScreen (): JSX.Element {
    return (
        <View style={tailwind('flex-1 items-center justify-center')}>
            <LoaderComponent size='large' />
        </View>
    )
}