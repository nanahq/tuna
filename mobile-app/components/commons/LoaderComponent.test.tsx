import {render} from '@testing-library/react-native'
import {LoaderComponent} from './LoaderComponent'

describe('<LoaderComponent />', () => {
    it('Should match snapshot', () => {
        const rendered = render(
          <LoaderComponent />
        )
        // @ts-ignore
        expect(rendered.toJSON()).toMatchSnapshot()
    })
})
