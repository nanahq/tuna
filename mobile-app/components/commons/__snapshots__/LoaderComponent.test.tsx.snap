// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoaderComponent /> Should match snapshot 1`] = `
<View
  style={
    [
      {},
      undefined,
    ]
  }
>
  <ActivityIndicator />
</View>
`;
